relay-filter/src/config.rs

```diff
@@ -328,15 +482,215 @@ impl FiltersConfig {
     }
 }
 
+/// Configuration of generic filters combined from project and global config.
+///
+/// See [`DynamicFiltersConfigIter`] for details on how to iterate easily
+/// through the applicable filters.
+#[derive(Copy, Clone, Debug)]
+pub(crate) struct DynamicFiltersConfig<'a> {
+    /// Configuration of generic filters from project configs.
+    project_filters: &'a GenericFiltersConfig,
+    /// Configuration of generic filters from global config.
+    global_filters: Option<&'a GenericFiltersConfig>,
+    /// Maximum supported version for generic filters.
+    ///
+    /// It applies to all filters, from both project and global configs.
+    max_version: u16,
+}
+
+impl<'a> DynamicFiltersConfig<'a> {
+    /// Creates a [`DynamicFiltersConfig`] from the project and global configs.
+    pub fn new(
+        project_filters: &'a GenericFiltersConfig,
+        global_filters: Option<&'a GenericFiltersConfig>,
+        max_version: u16,
```

Comment:
This confused me at first, this is the supported version, initially I thought it was the `global_version.max(project_version)`.

```suggestion
        supported_version: u16,
```